File: backend/internal/family/tree/vo/tree.go
package vo

type Node struct {
	MemberID            uint64  `json:"memberId"`
	DisplayName         string  `json:"displayName"`
	Surname             *string `json:"surname,omitempty"`
	GenerationCharacter *string `json:"generationCharacter,omitempty"`
	Gender              string  `json:"gender"`
	MemberType          string  `json:"memberType"`
	BirthDate           *string `json:"birthDate,omitempty"`
	DeathDate           *string `json:"deathDate,omitempty"`
	IsLiving            *bool   `json:"isLiving,omitempty"`
	UserBindingState    string  `json:"userBindingState"`
	CanExpand           bool    `json:"canExpand"`
	StopReason          *string `json:"stopReason,omitempty"`
}

type Edge struct {
	RelationshipID   uint64  `json:"relationshipId"`
	FromMemberID     uint64  `json:"fromMemberId"`
	ToMemberID       uint64  `json:"toMemberId"`
	RelationshipType string  `json:"relationshipType"`
	ParentLinkType   *string `json:"parentLinkType,omitempty"`
	RelationNoteType *string `json:"relationNoteType,omitempty"`
	RelationNote     *string `json:"relationNote,omitempty"`
}

type TreeItem struct {
	MemberID    uint64   `json:"memberId"`
	ParentIDs   []uint64 `json:"parentIds"`
	ChildrenIDs []uint64 `json:"childrenIds"`
	SpouseIDs   []uint64 `json:"spouseIds"`
}

type TreeResult struct {
	FamilyID     uint64     `json:"familyId"`
	TreeMode     string     `json:"treeMode"`
	GraphVersion int64      `json:"graphVersion"`
	Nodes        []Node     `json:"nodes"`
	Edges        []Edge     `json:"edges"`
	Tree         []TreeItem `json:"tree"`
}

File: web/src/api/tree.ts
import { apiClient, isRealApiMode } from '@/api/client'
import { treeEdges, treeNodes } from '@/mock/data'
import type { ApiResponse, FamilyTreeResult, RelationshipType } from '@/types/api'

export async function getPrivateTree(familyId: number | string): Promise<FamilyTreeResult> {
  if (!isRealApiMode) {
    const idByMockID = new Map(treeNodes.map((node, index) => [node.memberId, index + 1]))
    return {
      familyId: Number(familyId) || 1,
      treeMode: 'LIST_TREE',
      graphVersion: 1,
      nodes: treeNodes.map((node, index) => ({
        memberId: index + 1,
        displayName: node.name,
        gender: node.gender === '男' ? 'MALE' : node.gender === '女' ? 'FEMALE' : 'UNKNOWN',
        memberType: 'LINEAGE_MEMBER',
        birthDate: node.birthText ? `${node.birthText}-01-01` : undefined,
        deathDate: node.deathText ? `${node.deathText}-01-01` : undefined,
        isLiving: !node.deathText,
        userBindingState: 'UNBOUND',
        canExpand: true
      })),
      edges: treeEdges.map((edge, index) => ({
        relationshipId: index + 1,
        fromMemberId: idByMockID.get(edge.sourceMemberId) || 0,
        toMemberId: idByMockID.get(edge.targetMemberId) || 0,
        relationshipType: edge.relationshipType as RelationshipType
      })),
      tree: treeNodes.map((node, index) => ({
        memberId: index + 1,
        parentIds: node.parentIds.map((id) => idByMockID.get(id) || 0),
        childrenIds: node.childrenIds.map((id) => idByMockID.get(id) || 0),
        spouseIds: node.spouseIds.map((id) => idByMockID.get(id) || 0)
      }))
    }
  }
  const response = await apiClient.get<ApiResponse<FamilyTreeResult>>(`/families/${familyId}/tree`)
  return response.data.data
}
